File: Data Structures & Algorithms/three-integer-sum/submission-1.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        int length=nums.length;
        List<List<Integer>>ans_list = new ArrayList<>();

        Arrays.sort(nums);
        for(int i=0;i<length-2;i++){
            if(i>0 && nums[i]==nums[i-1]) continue;

            int left=i+1, right=length-1;
            while(left<right){
                int sum = nums[i]+nums[left]+nums[right];
                if(sum==0){
                    List<Integer>list = new ArrayList<>();
                    list.add(nums[i]);
                    list.add(nums[left]);
                    list.add(nums[right]);
                    
                    while(left<length-1 && nums[left]==nums[left+1]){
                        left++;
                    }
                    while(right>0 && nums[right]==nums[right-1]){
                        right--;
                    }
                    ans_list.add(list);
                    left++;
                    right--;
                }
                else if(sum<0){
                    left++;
                }
                else{
                    right--;
                }
            }
        }
        return ans_list;
    }
}

File: Data Structures & Algorithms/squares-of-a-sorted-array/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    sortedSquares(nums) {
        // return nums.map(ele=> ele**2).sort((a,b)=>a-b);

        let length=nums.length, ans=new Array(length);
        let left=0, right=length-1, ans_index=length-1;
        while(left<=right){
            if(Math.abs(nums[left])>Math.abs(nums[right])){
                ans[ans_index]=nums[left]**2;
                left++;
            }
            else{
                ans[ans_index]=nums[right]**2;
                right--;
            }
            ans_index--
        }
        return ans;
    }
}
